### PR TITLE
Automate S3 deployment

### PR DESCRIPTION
**Pull request by rosslo**

### CODE REVIEW

File: .github/workflows/s3-deploy-legacy.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Deploy legacy storybook to S3
```
Comment (rosslo) [inline .github/workflows/s3-deploy-legacy.yml:1]: This `yml` file is duplicated with `s3-deploy.yml` to temporarily support the S3 deployment for v4. 

So before I merge this MR, I will turn off buddy service first.

File: .github/workflows/s3-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: Deploy to S3
+on:
+  push:
+    branches:
+      - storybook6
```
Comment (rosslo) [inline .github/workflows/s3-deploy.yml:5]: will change this to `v4` after we merge the `storybook6` into `v4`.

File: .github/workflows/s3-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Build storybook
+        run: yarn storybook:build
+      - name: Upload storybook to S3
+        uses: jakejarvis/s3-sync-action@master
+        env:
+          AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_REGION: 'ap-southeast-1'
+          SOURCE_DIR: '.out'
```
Comment (rosslo) [inline .github/workflows/s3-deploy.yml:41]: So @westwood846 @ritikamotwani , actually our output dir is `.out` not `dist/public`. Sorry I didn't mindfully check the command arguments in package.json 😅

File: .github/workflows/s3-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: '10.x'
+      - name: Install dependencies
+        run: yarn install
```
Comment (rosslo) [inline .github/workflows/s3-deploy.yml:29]: The block from line 12 to here is copied from `cicd.yml`'s line 10 to 27 block.

File: .github/workflows/s3-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Build storybook
+        run: yarn storybook:build
+      - name: Upload storybook to S3
+        uses: jakejarvis/s3-sync-action@master
```
Comment (westwood846) [inline .github/workflows/s3-deploy.yml:33]: Might wanna lock this version and not use `master`.

File: .github/workflows/s3-deploy.yml (hunk 3 above)
Comment (westwood846) [reply .github/workflows/s3-deploy.yml:41]: Oooh good to know!

**Review by westwood846 — APPROVED** [review]
Well there isn't much code here to review actually ^^ This is approved.

File: .github/workflows/s3-deploy.yml (hunk 5 above)
Comment (rosslo) [reply .github/workflows/s3-deploy.yml:33]: Looks like there is no a fixed version tag can be used from their document. 🤔

**Pull request merged**